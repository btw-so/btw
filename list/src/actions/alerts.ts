import React from 'react';
import { uuid } from '@gilbarbara/helpers';
import { createAction } from '@reduxjs/toolkit';

import { actionPayload } from 'modules/helpers';

import { ActionTypes } from 'literals';

import { ShowAlertOptions } from 'types';

export const hideAlert = createAction<string>(ActionTypes.HIDE_ALERT);

export const showAlert = createAction(
  ActionTypes.SHOW_ALERT,
  (message: React.ReactNode, options: ShowAlertOptions) => {
    const timeout = options.variant === 'danger' ? 0 : 5;

    return actionPayload({
      id: options.id || uuid(),
      icon: options.icon || 'dot-circle-o',
      message,
      position: options.position || 'bottom-right',
      variant: options.variant || 'dark',
      timeout: typeof options.timeout === 'number' ? options.timeout : timeout,
    });
  },
);
