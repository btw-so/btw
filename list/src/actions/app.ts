import { createAction } from '@reduxjs/toolkit';

import { ActionTypes } from 'literals';

import { SetAppOOptions } from 'types';

export const setAppOptions = createAction<SetAppOOptions>(ActionTypes.SET_APP_OPTIONS);
