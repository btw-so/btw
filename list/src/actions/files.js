import { createAction } from "@reduxjs/toolkit";

import { ActionTypes } from "literals";


export const addFile = createAction(
  ActionTypes.ADD_FILE,
  (data) => ({
    payload: data,
  })
);
export const addFileSuccess = createAction(
  ActionTypes.ADD_FILE_SUCCESS,
  (data) => ({
    payload: data,
  })
);
export const addFileFailure = createAction(
  ActionTypes.ADD_FILE_FAILURE,
  (data) => ({
    payload: data,
  })
);

export const getFile = createAction(ActionTypes.GET_FILE, (data) => ({
  payload: data,
}));
export const getFileSuccess = createAction(
  ActionTypes.GET_FILE_SUCCESS,
  (data) => ({
    payload: data,
  })
);
export const getFileFailure = createAction(
  ActionTypes.GET_FILE_FAILURE,
  (data) => ({
    payload: data,
  })
);
