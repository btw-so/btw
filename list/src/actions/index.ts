export * from './alerts';
export * from './app';
export * from './github';
export * from './user';
export * from './list';
export * from './notes';