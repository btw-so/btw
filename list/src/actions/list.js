import { createAction } from "@reduxjs/toolkit";

import { ActionTypes } from "literals";

export const batchPushNodes = createAction(
  ActionTypes.BATCH_PUSH_NODES,
  (data) => ({
    payload: data,
  })
);

export const getList = createAction(ActionTypes.GET_LIST, (data) => ({
  payload: data,
}));
export const getListSuccess = createAction(
  ActionTypes.GET_LIST_SUCCESS,
  (data) => ({
    payload: data,
  })
);
export const getListFailure = createAction(
  ActionTypes.GET_LIST_FAILURE,
  (data) => ({
    payload: data,
  })
);

export const newListNode = createAction(ActionTypes.NEW_LIST_NODE, (data) => ({
  payload: data,
}));

export const editListNodePos = createAction(
  ActionTypes.EDIT_LIST_NODE_POS,
  (data) => ({
    payload: data,
  })
);

export const editListNodeContent = createAction(
  ActionTypes.EDIT_LIST_NODE_CONTENT,
  (data) => ({
    payload: data,
  })
);

export const upsertListNode = createAction(
  ActionTypes.UPSERT_LIST_NODE,
  (data) => ({
    payload: data,
  })
);

export const changeSelectedNode = createAction(
  ActionTypes.CHANGE_SELECTED_NODE,
  (data) => ({
    payload: data,
  })
);

export const getPinnedNodes = createAction(
  ActionTypes.GET_PINNED_NODES,
  (data) => ({
    payload: data,
  })
);

export const getPinnedNodesSuccess = createAction(
  ActionTypes.GET_PINNED_NODES_SUCCESS,
  (data) => ({
    payload: data,
  })
);

export const getPinnedNodesFailure = createAction(
  ActionTypes.GET_PINNED_NODES_FAILURE,
  (data) => ({
    payload: data,
  })
);

export const getPublicNote = createAction(
  ActionTypes.GET_PUBLIC_NOTE,
  (data) => ({
    payload: data,
  })
);

export const getPublicNoteSuccess = createAction(
  ActionTypes.GET_PUBLIC_NOTE_SUCCESS,
  (data) => ({
    payload: data,
  })
);

export const getPublicNoteFailure = createAction(
  ActionTypes.GET_PUBLIC_NOTE_FAILURE,
  (data) => ({
    payload: data,
  })
);

export const searchNodes = createAction(
  ActionTypes.SEARCH_NODES,
  (data) => ({
    payload: data,
  })
);

export const searchNodesSuccess = createAction(
  ActionTypes.SEARCH_NODES_SUCCESS,
  (data) => ({
    payload: data,
  })
);

export const searchNodesFailure = createAction(
  ActionTypes.SEARCH_NODES_FAILURE,
  (data) => ({
    payload: data,
  })
);
