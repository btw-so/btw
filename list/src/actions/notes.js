import { createAction } from "@reduxjs/toolkit";

import { ActionTypes } from "literals";


export const saveNoteContent = createAction(
  ActionTypes.SAVE_NOTE_CONTENT,
  (data) => ({
    payload: data,
  })
);

export const getNote = createAction(ActionTypes.GET_NOTE, (data) => ({
  payload: data,
}));
export const getNoteSuccess = createAction(
  ActionTypes.GET_NOTE_SUCCESS,
  (data) => ({
    payload: data,
  })
);
export const getNoteFailure = createAction(
  ActionTypes.GET_NOTE_FAILURE,
  (data) => ({
    payload: data,
  })
);
