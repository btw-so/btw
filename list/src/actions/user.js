import { createAction } from "@reduxjs/toolkit";

import { ActionTypes } from "literals";

export const login = createAction(ActionTypes.USER_LOGIN_REQUEST);
export const loginSuccess = createAction(ActionTypes.USER_LOGIN_SUCCESS);

export const logOut = createAction(ActionTypes.USER_LOGOUT_REQUEST);
export const logOutSuccess = createAction(ActionTypes.USER_LOGOUT_SUCCESS);

export const getUser = createAction(ActionTypes.GET_USER);
export const getUserSuccess = createAction(
  ActionTypes.GET_USER_SUCCESS,
  (data) => ({
    payload: data,
  })
);
export const getUserFailure = createAction(
  ActionTypes.GET_USER_FAILURE,
  (data) => ({
    payload: data,
  })
);

export const generateOtp = createAction(ActionTypes.GENERATE_OTP, (data) => ({
  payload: data,
}));
export const generateOtpSuccess = createAction(
  ActionTypes.GENERATE_OTP_SUCCESS,
  ({ success, error, data }) => ({ payload: { success, error, data } })
);
export const generateOtpFailure = createAction(
  ActionTypes.GENERATE_OTP_FAILURE,
  (data) => ({
    payload: data,
  })
);

export const verifyOtp = createAction(
  ActionTypes.VERIFY_OTP,
  ({ email, otp } = {}) => ({
    payload: { email, otp },
  })
);
export const verifyOtpSuccess = createAction(
  ActionTypes.VERIFY_OTP_SUCCESS,
  (data) => ({
    payload: data,
  })
);
export const verifyOtpFailure = createAction(
  ActionTypes.VERIFY_OTP_FAILURE,
  (data) => ({
    payload: data,
  })
);

export const updateUser = createAction(
  ActionTypes.UPDATE_USER,
  ({ name, slug, bio, pic, twitter, linkedin, instagram, settings } = {}) => ({
    payload: { name, slug, bio, pic, twitter, linkedin, instagram, settings },
  })
);
export const updateUserSuccess = createAction(
  ActionTypes.UPDATE_USER_SUCCESS,
  (data) => ({
    payload: data,
  })
);
export const updateUserFailure = createAction(
  ActionTypes.UPDATE_USER_FAILURE,
  (data) => ({
    payload: data,
  })
);
export const resetState = createAction(
  ActionTypes.RESET_USER_STATE,
  (data) => ({
    payload: data,
  })
);
