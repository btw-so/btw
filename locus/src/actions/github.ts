import { createAction } from '@reduxjs/toolkit';

import { actionPayload } from 'modules/helpers';

import { ActionTypes } from 'literals';

export interface GetReposSuccessMeta {
  cached: boolean;
  query: string;
  updatedAt: number;
}

export const getRepos = createAction<string>(ActionTypes.GITHUB_GET_REPOS_REQUEST);
export const getReposSuccess = createAction(
  ActionTypes.GITHUB_GET_REPOS_SUCCESS,
  (payload: Record<string, any>[], meta: GetReposSuccessMeta) => actionPayload(payload, meta),
);
export const getReposFailure = createAction(
  ActionTypes.GITHUB_GET_REPOS_FAILURE,
  (payload: string, query: string) => actionPayload(payload, { query }),
);
