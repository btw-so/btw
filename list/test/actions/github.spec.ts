import { getRepos } from 'actions/github';

describe('actions/github', () => {
  it('getRepos', () => {
    expect(getRepos('react')).toMatchSnapshot();
  });
});
