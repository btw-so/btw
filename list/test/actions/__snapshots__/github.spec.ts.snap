// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions/github getRepos 1`] = `
Object {
  "payload": "react",
  "type": "GITHUB_GET_REPOS_REQUEST",
}
`;
